webpack/JobWizard/steps/HostsAndInputs/HostPreviewModal.js

```diff
@@ -0,0 +1,62 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { useSelector } from 'react-redux';
+import URI from 'urijs';
+import { List, ListItem, Modal, Button } from '@patternfly/react-core';
+import { translate as __, sprintf } from 'foremanReact/common/I18n';
+import { foremanUrl } from 'foremanReact/common/helpers';
+import { selectHosts, selectHostCount } from '../../JobWizardSelectors';
+import { HOSTS_TO_PREVIEW_AMOUNT } from '../../JobWizardConstants';
+
+export const HostPreviewModal = ({ isOpen, setIsOpen, searchQuery }) => {
+  const hosts = useSelector(selectHosts);
+  const hostsCount = useSelector(selectHostCount);
+  const url = new URI(foremanUrl('/hosts'));
+
+  return (
+    <Modal
+      title={__('Preview Hosts')}
+      isOpen={isOpen}
+      onClose={() => setIsOpen(false)}
+      appendTo={() => document.getElementsByClassName('job-wizard')[0]}
+    >
+      <List isPlain>
+        {hosts.map(host => (
+          <ListItem key={host}>
+            <Button
+              component="a"
+              href={foremanUrl(`/hosts/${host}`)}
+              variant="link"
+              target="_blank"
+              rel="noreferrer"
+            >
+              {host}
+            </Button>
+          </ListItem>
+        ))}
+        {hostsCount > HOSTS_TO_PREVIEW_AMOUNT && (
+          <ListItem>
+            <Button
+              component="a"
+              href={url.addSearch({ search: searchQuery })}
+              variant="link"
+              target="_blank"
+              rel="noreferrer"
+            >
+              {sprintf(
+                __('...and %s more'),
+                hostsCount - HOSTS_TO_PREVIEW_AMOUNT
+              )}
+            </Button>
+          </ListItem>
+        )}
+      </List>
+    </Modal>
```

Comment:
how does the list look with many hosts? can you share a screenshot?

Reply:
for 101 hosts
![Screenshot from 2021-10-18 16-45-35](https://user-images.githubusercontent.com/30431079/137754156-98e80057-b6e3-4dea-8a5e-2bef3afe6174.png)